### PR TITLE
Allow fails in Remove large packages

### PR DESCRIPTION
**Pull request by kfir4444**
Added `|| true` at the end of large file removal, so that if one step of it fails, the clean-up job would continue.

### CODE REVIEW

**Comment by kfir4444** [comment]
Thanks for the replay @jlumbroso! 😃 
Regarding your first question, I can think of some scenarios where this action could be used as a test, so a failure here could be important, but I think that the general use case only benefits from not stopping a CI due to failed deletion, which hinders development.
Well, we use this github action in (at least) one of our [repos](https://github.com/ReactionMechanismGenerator/ARC) and got some CI fails ([example here](https://github.com/ReactionMechanismGenerator/ARC/actions/runs/5276764998/jobs/9545168360)) due to a false return code in ubuntu. I think that the users can tolerate a difference of a few mb's extra, but have a CI fail a few times due to uninstallation error may discourage users to use this tool. Also, I recall talking to a colleague and postulating that this issue may had something to do with the new ubuntu release and additional packages still being adapted to it (could not be verified, though), so this PR (which is already implemented in our software for a while using my fork) gave us the opportunity to continue developing even in a (somewhat) unstable environment associated with open source code (which we absolutely love!).

**Comment by pratik-techholding** [comment]
Solution of @kfir4444 looks good can we approve this PR please many user facing issues

**Comment by kfir4444** [comment]
Thanks @pratik-techholding!
I do not think we should hate the code review process, if you want, you can use [my clone](https://github.com/kfir4444/free-disk-space/tree/main) until this PR is merge. I can ping you when it's merged 😄

**Comment by AurevoirXavier** [comment]
Maybe this could fix #9.

File: action.yml
Comment (ChrisCarini) [inline action.yml:176]:
```suggestion
          sudo apt-get remove -y '^dotnet-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^dotnet-.*' --fix-missing] failed to complete successfully. Proceeding..."
          sudo apt-get remove -y '^llvm-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^llvm-.*' --fix-missing] failed to complete successfully. Proceeding..."
          sudo apt-get remove -y 'php.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y 'php.*' --fix-missing] failed to complete successfully. Proceeding..."
          sudo apt-get remove -y '^mongodb-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^mongodb-.*' --fix-missing] failed to complete successfully. Proceeding..."
          sudo apt-get remove -y '^mysql-.*' --fix-missing || echo "::warning::The command [sudo apt-get remove -y '^mysql-.*' --fix-missing] failed to complete successfully. Proceeding..."
          sudo apt-get remove -y azure-cli google-cloud-sdk google-chrome-stable firefox powershell mono-devel libgl1-mesa-dri --fix-missing || echo "::warning::The command [sudo apt-get remove -y azure-cli google-cloud-sdk google-chrome-stable firefox powershell mono-devel libgl1-mesa-dri --fix-missing] failed to complete successfully. Proceeding..."
          sudo apt-get autoremove -y || echo "::warning::The command [sudo apt-get autoremove -y] failed to complete successfully. Proceeding..."
          sudo apt-get clean || echo "::warning::The command [sudo apt-get clean] failed to complete successfully. Proceeding..."
```

**Review by ChrisCarini — APPROVED** [review]
🎉 LGTM! Thank you for this change!

I'd be happy if this PR is merged as-is to simply unblock about a dozen or so of my repos' CI runs. The suggestion simply addresses @jlumbroso's question in [their comment here](https://github.com/jlumbroso/free-disk-space/pull/8#issuecomment-1611926548); IMO, having messages printed for these command failures a 'nice-to-have', but not required. I'd rather see my CI jobs unblocked sooner rather than later.


@jlumbroso - by allowing this PR to be merged in and released, the below issues / PRs would be closed / addressed:
- #9
- #11 

Thoughts on getting this merged in to unblock the many CI runs that are affected by issues this PR addresses?

**Comment by yinweisu** [comment]
Hi, when do we expect to get the fix merged in?

**Comment by ChrisCarini** [comment]
Hi @jlumbroso - do you have any subsequent concerns and/or blockers to getting this PR merged?

If you need extra help maintaining this repository, I'd be happy to volunteer to assist (I have experience with authoring & maintaining several GHAs); just let me know.

**Comment by kfir4444** [comment]
@jlumbroso as @ChrisCarini, I can also help with maintenance :smile:

**Comment by kfir4444** [comment]
For everyone else struggling with issues, you can use my fork for now (this is what we do in our repos) until this chage becomes official. I can tag you here when this PR is merged so you can return to the original.
@AurevoirXavier @yinweisu @ChrisCarini

**Comment by causten** [comment]
> Thanks for the replay @jlumbroso! 😃 Regarding your first question, I can think of some scenarios where this action could be used as a test, so a failure here could be important, but I think that the general use case only benefits from not stopping a CI due to failed deletion, which hinders development. 

Don't you get for free the error check if you set "continue-on-error: true" in the action?

**Comment by Stephane-Ag** [comment]
This would also fix #4. I assume @jlumbroso has just not had a chance to look at this.

**Review by jlumbroso — APPROVED** [review]

**Comment by jlumbroso** [comment]
Sorry for the delay. @ChrisCarini Thank you for addressing my concerns with your change.

**Comment by ChrisCarini** [comment]
> Sorry for the delay. 

No worries!

> @ChrisCarini Thank you for addressing my concerns with your change.

Of course! FWIW, my offer to help you maintain this repo (ref: https://github.com/jlumbroso/free-disk-space/pull/8#issuecomment-1685200176 ) (approving/merging PRs, releasing new versions, etc) still stands. :) Just let me know! Happy to help.

**Pull request merged**
